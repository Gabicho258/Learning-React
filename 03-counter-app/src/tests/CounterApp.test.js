import React from 'react'
import "@testing-library/jest-dom"
import {shallow} from "enzyme"
import CounterApp from "../CounterApp";

describe('Tarea', () => {
    let wrapper = shallow(<CounterApp/>); 
    beforeEach(() => {
        wrapper = shallow(<CounterApp/>)
        
    })
    test('Debe mostrar correctamente counter app', () => {
        // const valueDefault = 10;
        const wrapper = shallow(<CounterApp />)
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe mostrar el valor por defecto', () => {
        const valueDefault = 100;
        const wrapper = shallow(
            <CounterApp 
            value = {valueDefault}
            />
            )
            const text = parseInt(wrapper.find("h2").text());
            expect(text).toBe(valueDefault);
            
        })
    test('Debe incrementar con el boton +1', () => {
        wrapper.find("button").at(0).simulate('click'); 
        const counterText = wrapper.find("h2").text().trim();
        expect(counterText).toBe("1")
    })
    test('Debe decrementar con el boton -1', () => {
        wrapper.find("button").at(2).simulate('click'); 
        const counterText = wrapper.find("h2").text().trim();
        expect(counterText).toBe("-1")
    })
    test('Debe resetearse con el boton reset', () => {
        const valueDefault = 100;
        const wrapper = shallow(
            <CounterApp 
            value = {valueDefault}
            />
            )
        wrapper.find("button").at(0).simulate('click'); 
        wrapper.find("button").at(1).simulate('click'); 
        const counterText = wrapper.find("h2").text().trim();

        expect(counterText).toBe(`${valueDefault}`)
    })
    
    
})
