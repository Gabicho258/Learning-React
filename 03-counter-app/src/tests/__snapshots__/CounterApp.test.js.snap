// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tarea Debe incrementar con el boton +1 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;

exports[`Tarea Debe mostrar correctamente counter app 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
